pkg/github/github_client.go

```diff
@@ -80,32 +83,69 @@ func newGithubClientByApp(appId int64, privateKeyPem []byte, owner string) (*Git
 	return NewGithubClient(token.GetToken()), nil
 }
 
-func (c *GithubClient) GetReference(owner, repository, branch string) (*github.Reference, error) {
+func (c *GithubClient) isReferenceExist(owner, repository, branch string) (bool, error) {
+	_, resp, err := c.client.Git.GetRef(c.ctx, owner, repository, "refs/heads/"+branch)
+	if err == nil {
+		return true, nil
+	}
+
+	if resp.StatusCode == 404 {
+		return false, nil
+	}
+	return false, err
+}
+
+func (c *GithubClient) getReference(owner, repository, branch string) (*github.Reference, error) {
 	ref, _, err := c.client.Git.GetRef(c.ctx, owner, repository, "refs/heads/"+branch)
 	return ref, err
 }
 
-func (c *GithubClient) GetOrCreateBranchReference(owner, repository, branch, baseBranch string) (*github.Reference, error) {
-	ref, resp, err := c.client.Git.GetRef(c.ctx, owner, repository, "refs/heads/"+branch)
-	if err == nil {
-		return ref, nil
-	} else if resp.StatusCode != 404 {
-		return nil, err
-	}
-
-	baseBranchRef, err := c.GetReference(owner, repository, baseBranch)
+func (c *GithubClient) createReference(owner, repository, branch, baseBranch string) (*github.Reference, error) {
+	baseBranchRef, err := c.getReference(owner, repository, baseBranch)
 	if err != nil {
 		return nil, err
 	}
 	newBranchRef := &github.Reference{
 		Ref:    github.String("refs/heads/" + branch),
 		Object: &github.GitObject{SHA: baseBranchRef.Object.SHA},
 	}
-	ref, _, err = c.client.Git.CreateRef(c.ctx, owner, repository, newBranchRef)
+	ref, _, err := c.client.Git.CreateRef(c.ctx, owner, repository, newBranchRef)
 	return ref, err
 }
 
-func (c *GithubClient) CreateTree(owner, repository string, baseRef *github.Reference, files []File) (tree *github.Tree, err error) {
+func (c *GithubClient) deleteReference(owner, repository, branch string) error {
+	_, err := c.client.Git.DeleteRef(c.ctx, owner, repository, "refs/heads/"+branch)
+	return err
+}
+
+func (c *GithubClient) isFilesUpToDate(owner, repository, branch string, files []File) (bool, error) {
```

Comment:
another nitpick: I guess it should be `areFilesUpToDate`. (unless there's some naming schema I'm not aware of)